hrpc/call.go

```diff
@@ -110,7 +111,9 @@ type RPCResult struct {
 }
 
 type base struct {
-	ctx     context.Context
+	ctxM sync.Mutex
```

Comment:
This is a symptom of the problem Warren raised when he added tracing support. We shouldn't store the context in the RPC object, we should pass it along, and when we derive a new context due to tracing we just pass along the derived context.

Adding this here is fine for now because it fixes a real bug, but the proper fix would be to remove context from this struct altogether. I filed #174 to track this.